packages/sdk/src/services/wallet.service.ts

```diff
@@ -0,0 +1,224 @@
+          chain: holographToViemChain(network.chain),
+          account: this._account,
+          transport: http(network.rpc),
+        }).extend(publicActions)
+      })
+    }
+  }
+
+  get account() {
+    return this._account
+  }
+
+  onChain(chainId: number): WalletClient {
+    const logger = this._logger.addContext({functionName: this.onChain.name})
+    logger.info(`wallet client accessing chainId = ${chainId}`)
+
+    const walletClient = this._multiChainWalletClient[chainId]
+
+    if (walletClient === undefined) {
+      throw new UnavailableNetworkError(chainId, this.onChain.name)
+    }
+
+    return walletClient
+  }
+}
+
+/**
+ * HolographWalletManager
+ *
+ * @remarks
+ *
+ * Manages a set of HolographWallets.
+ *
+ */
+export class HolographWalletManager {
+  private readonly _logger: HolographLogger
+  private readonly _networks: Network[]
+  private _wallets: {[accountName: string]: HolographWallet} = {}
+  private _addressToAccountName: {[address: Address]: string} = {}
+
+  constructor(private readonly protocolConfig: Config) {
+    this._logger = HolographLogger.createLogger({serviceName: HolographWalletManager.name})
+
+    if (this.protocolConfig.accounts?.default === undefined) {
+      throw new Error('No default wallet configured')
+    }
+
+    this._networks = this.protocolConfig.networks
+
+    for (const [accountName, account] of Object.entries(this.protocolConfig.accounts)) {
+      this._wallets[accountName] = new HolographWallet({account, networks: this._networks})
+      this._addressToAccountName[account.address] = accountName
+    }
+  }
+
+  get allWallets(): {[accountName: string]: HolographWallet} {
+    return this._wallets
+  }
+
+  getWallet(account: (string & keyof typeof this._wallets) | Address = 'default') {
+    let accountName: string | undefined = account
+
+    if (isAddress(account)) {
+      accountName = this._addressToAccountName[account]
+    }
+
+    if (Object.keys(this._wallets).includes(accountName)) {
+      return this._wallets[accountName]
+    }
+
+    throw new Error(`No wallet was found for ${account}.`)
+  }
+
+  getAccountNameToAddressMap(): {[accountName: string]: Address} {
+    const map = {}
+
+    for (const [accountName, wallet] of Object.entries(this._wallets)) {
+      map[accountName] = wallet.account
+    }
+
+    return map
+  }
+
+  addAccount(accountName: string, account: HolographAccount): HolographWallet {
+    const logger = this._logger.addContext({functionName: this.addAccount.name})
+
+    let name = accountName
+    if (this._wallets === undefined) {
+      logger.warn(`Setting the "default" account to ${account}`)
+      name = 'default'
+    }
+
+    const notAllowed = Object.keys(this._wallets)
+
+    if (notAllowed.includes(accountName)) {
+      throw new Error('The chosen account name is already in use.')
+    }
+
+    this._wallets[name] = new HolographWallet({account, networks: this._networks})
+    return this._wallets[name]
```

Comment:
May be good to mention the existance of `updateAccount` fn. So maybe change copy to: "The provided account name is already in use. Please use updateAccount to override the name"